src/TestSuite/MiddlewareDispatcher.php

```diff
@@ -164,7 +164,7 @@ protected function _createRequest($spec)
             $spec['post'] = [];
         }
         $environment = array_merge(
-            array_merge($_SERVER, ['REQUEST_URI' => $spec['url'], 'PHP_SELF' => '/']),
+            array_merge(['REQUEST_URI' => $spec['url'], 'PHP_SELF' => '/'], $_SERVER),
```

Comment:
Won't this result in REQUEST_URI being wrong? That could explain some of the test failures.

Reply:
Only if REQUEST_URI is also being specified in the bootstrap or elsewhere, which seemed unlikely. (And if it is, maybe it should take precedence? I don't know.) Tried to run the tests first on my server, but got many failures, likely due to unsupported versions of things like SQLite. Will come back to this a little later, when I have more time to really dig into it.

Reply:
Latest commit resolves this. Any problems with it, or just haven't had time to review?

Reply:
I had missed the update notification on this.